cmd/yggdrasil/main.go

```diff
@@ -188,6 +193,26 @@ func main() {
 	if cfg == nil {
 		return
 	}
+	// Have we been asked for the node address yet? If so, print it and then stop.
+	switch {
+	case *getaddr:
+		if pubkey, err := hex.DecodeString(cfg.EncryptionPublicKey); err == nil {
+			nodeid := sha512.Sum512(pubkey)
```

Comment:
It would be better to use something like `crypto.GetNodeID(crypto.BoxPubKey(pubkey))` instead of importing and directly using sha512 here. The details of the key->ID->address mapping should be opaque, in case we ever need to change them for some reason (e.g. post-quantum encryption).